walletkit_client.go

```diff
@@ -407,3 +458,135 @@ func (m *walletKitClient) ListAccounts(ctx context.Context, name string,
+		return nil, 0, nil, err
+	}
+
+	packet, err := psbt.NewFromRawBytes(
+		bytes.NewReader(resp.FundedPsbt), false,
+	)
+	if err != nil {
+		return nil, 0, nil, err
+	}
+
+	return packet, resp.ChangeOutputIndex, resp.LockedUtxos, nil
+
+}
+
+// SignPsbt expects a partial transaction with all inputs and outputs
+// fully declared and tries to sign all unsigned inputs that have all
+// required fields (UTXO information, BIP32 derivation information,
+// witness or sig scripts) set.
+// If no error is returned, the PSBT is ready to be given to the next
+// signer or to be finalized if lnd was the last signer.
+//
+// NOTE: This RPC only signs inputs (and only those it can sign), it
+// does not perform any other tasks (such as coin selection, UTXO
+// locking or input/output/fee value validation, PSBT finalization). Any
+// input that is incomplete will be skipped.
+func (m *walletKitClient) SignPsbt(ctx context.Context,
+	packet *psbt.Packet) (*psbt.Packet, error) {
+
+	var psbtBuf bytes.Buffer
+	if err := packet.Serialize(&psbtBuf); err != nil {
+		return nil, err
+	}
+
+	rpcCtx, cancel := context.WithTimeout(ctx, m.timeout)
+	defer cancel()
+
+	resp, err := m.client.SignPsbt(
+		m.walletKitMac.WithMacaroonAuth(rpcCtx),
+		&walletrpc.SignPsbtRequest{FundedPsbt: psbtBuf.Bytes()},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	signedPacket, err := psbt.NewFromRawBytes(
+		bytes.NewReader(resp.SignedPsbt), false,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return signedPacket, nil
+}
+
+// FinalizePsbt expects a partial transaction with all inputs and
+// outputs fully declared and tries to sign all inputs that belong to
+// the wallet. Lnd must be the last signer of the transaction. That
+// means, if there are any unsigned non-witness inputs or inputs without
+// UTXO information attached or inputs without witness data that do not
+// belong to lnd's wallet, this method will fail. If no error is
+// returned, the PSBT is ready to be extracted and the final TX within
+// to be broadcast.
+//
+// NOTE: This method does NOT publish the transaction once finalized. It
+// is the caller's responsibility to either publish the transaction on
+// success or unlock/release any locked UTXOs in case of an error in
+// this method.
+func (m *walletKitClient) FinalizePsbt(ctx context.Context, packet *psbt.Packet,
+	account string) (*psbt.Packet, *wire.MsgTx, error) {
+
+	var psbtBuf bytes.Buffer
+	if err := packet.Serialize(&psbtBuf); err != nil {
+		return nil, nil, err
+	}
+
+	rpcCtx, cancel := context.WithTimeout(ctx, m.timeout)
+	defer cancel()
+
+	resp, err := m.client.FinalizePsbt(
+		m.walletKitMac.WithMacaroonAuth(rpcCtx),
+		&walletrpc.FinalizePsbtRequest{
+			FundedPsbt: psbtBuf.Bytes(),
+			Account:    account,
+		},
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	finalizedPacket, err := psbt.NewFromRawBytes(
+		bytes.NewReader(resp.SignedPsbt), false,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	finalTx := wire.NewMsgTx(2)
```

Comment:
out of scope for this PR but we may want to create some iota for the tx version?

Reply:
Hmm, it looks like we're only using it this way throughout the lnd codebase. But you're right, we should probably do it at some point :)